Saving a node deletion that has been modified externally throws a ConstraintViolationException
Deleting a node a and saving its parent might result in a ConstraintViolationException if node a has been modified externally where an InvalidItemStateException with message item x has been modified externally would be more intuitive. Test case exposing issue. What happens is session2 removes a session1 adds child b to a session1 saves session2 saves The last line throws a ConstraintViolationException because save detects a child node b in the local state that does not exist in the transient state and which it therefore consideres removed.fixed in r633179Merged to the 1.4 branch in revision 633862.
